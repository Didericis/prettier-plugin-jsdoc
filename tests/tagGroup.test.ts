import prettier from "prettier";
import { AllOptions } from "../src/types";

function subject(code: string, options: Partial<AllOptions> = {}) {
  return prettier.format(code, {
    plugins: ["."],
    ...options,
  } as AllOptions);
}

test("Tag group", () => {
  const result = subject(
    `
  /**
   * Aliquip ex proident tempor eiusmod aliquip amet. Labore commodo nulla tempor
   * consequat exercitation incididunt non. Duis laboris reprehenderit proident
   * proident.
   * @see {@link http://acme.com}
   * @example
   *   const foo = 0;
   *   const a = "";
   *   const b = "";
   *
   * @param id A test id.
   * @throws Minim sit ad commodo ut dolore magna magna minim consequat. Ex
   *   consequat esse incididunt qui voluptate id voluptate quis ex et. Ullamco
   *   cillum nisi amet fugiat.
   * @return Minim sit a.
   */
  
`,
    {
      jsdocSeparateTagGroups: true,
    },
  );

  expect(result).toMatchSnapshot();
});
