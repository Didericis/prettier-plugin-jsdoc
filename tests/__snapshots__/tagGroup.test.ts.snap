// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag group 1`] = `
"/**
 * Aliquip ex proident tempor eiusmod aliquip amet. Labore commodo nulla
 * tempor consequat exercitation incididunt non. Duis laboris reprehenderit
 * proident proident.
 *
 * @example
 *   const foo = 0;
 *   const a = "";
 *   const b = "";
 *
 * @param id A test id.
 *
 * @returns Minim sit a.
 *
 * @throws Minim sit ad commodo ut dolore magna magna minim consequat. Ex
 *   consequat esse incididunt qui voluptate id voluptate quis ex et. Ullamco
 *   cillum nisi amet fugiat.
 *
 * @see {@link http://acme.com}
 */
"
`;
